graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -186,25 +238,28 @@ class RenameSchemaTypesVisitor(Visitor):
         }
     )
 
-    def __init__(self, renamings, query_type, scalar_types):
+    def __init__(
+        self, renamings: Dict[str, Optional[str]], query_type: str, scalar_types: Set[str]
+    ) -> None:
         """Create a visitor for renaming types in a schema AST.
 
         Args:
-            renamings: Dict[str, str], mapping from original type name to renamed type name.
-                       Any name not in the dict will be unchanged
+            renamings: Dict[str, Optional[str]], mapping from original type name to renamed type
+                       name or None (for type suppression). Any name not in the dict will be
+                       unchanged
             query_type: str, name of the query type (e.g. RootSchemaQuery), which will not
                         be renamed
             scalar_types: Set[str], set of names of all scalars used in the schema, including
                           all user defined scalars and any builtin scalars that were used
         """
         self.renamings = renamings
-        self.reverse_name_map = {}  # Dict[str, str], from renamed type name to original type name
-        # reverse_name_map contains all types, including those that were unchanged
+        self.reverse_name_map: Dict[str, str] = {}  # From renamed type name to original type name
+        # reverse_name_map contains all non-suppressed types, including those that were unchanged
         self.query_type = query_type
         self.scalar_types = frozenset(scalar_types)
         self.builtin_types = frozenset({"String", "Int", "Float", "Boolean", "ID"})
 
-    def _rename_name_and_add_to_record(self, node):
+    def _rename_name_and_add_to_record(self, node: Node) -> Union[type(REMOVE), Optional[Node]]:
```

Comment:
Can this function return `None`? The `Returns` section of the docstring seems to say otherwise.

Reply:
Thanks for the catch! I've looked more closely into how the library defines special values other than `REMOVE`. It seems like everywhere the code previously returned `None`, we should return `IDLE` instead (which happens to be set to `None` right now).

In that case, based on the issue I opened in GraphQL-core today, I'll adjust `VisitorReturnType` (as mentioned in a previous comment above) to be the union of `Node` and `Any`, and when the special return values get collected into an enum it'll simply be a matter of changing the `Any` instead of messing around with optional types.